### ISSUE TITLE
Wrong Yandex Mail settings by default

### ISSUE THREAD

**Issue by westsibe** [deltachat/deltachat-android#1526]
Delta Chat for Android.

imap.yandex.ru (incoming server) and smtp.yandex.ru (outgoing server) is always for login@ya.ru, login@yandex.ru, login@yandex.com and login@yandex.ua! 
No others smtp and imap servers for Yandex Mail!

2. But and with this custom imap settings Yandex Mail not work a long time ago in DC for Android...:(

(P.  S. Yandex it is No.1 as free e-mail in Russia and No.2 as Web search.)

**Comment by link2xt** [comment]
It is a bug in provider database.

Compare https://raw.githubusercontent.com/deltachat/provider-db/master/_providers/gmail.md and https://raw.githubusercontent.com/deltachat/provider-db/master/_providers/yandex.ru.md

Yandex does not have servers specified at all.

**Comment by westsibe** [comment]
IMAP protocol is on there, and e-mail clients for Android (AquaMail for example) is works with Yandex Mail.
(Oauth in DC also not work with Yandex Mail.)

**Comment by gerryfrancis** [comment]
`strict_tls` is set `true` in https://github.com/deltachat/provider-db/blob/master/_providers/yandex.ru.md , I guess it should be removed.

**Comment by link2xt** [comment]
Why? Yandex has valid certificates.

**Comment by gerryfrancis** [comment]
@link2xt Ah, I thought this feature was valid only for STARTTLS. Sorry about that.

But something else that is mentioned on the [Yandex IMAP configuration support site](https://yandex.com/support/mail/mail-clients.html) caught my eye:

`If you are configuring email delivery from a mailbox such as “login@yandex.com”, your username is the first part of the address before the “@” sign.`

That would be `username_pattern = EMAILLOCALPART` , right?

**Comment by link2xt** [comment]
A bit off-topic, related bug.

> But and with this custom imap settings Yandex Mail not work a long time ago in DC for Android...:(

I have just tested, with `@yandex.ru` and OAuth2 it works on Android, but you have to enable IMAP in Yandex Mail web interface first. Look into the logs for lines about bad IMAP response (`[CLIENTBUG]`), if it is there, then authentication actually failed, but it was not reported to you due to https://github.com/async-email/async-imap/issues/35

Sure providers database still should be fixed to make other aliases like `@ya.ru` work. I was just testing `@yandex.ru` first and found this async-imap bug.

**Comment by gerryfrancis** [comment]
> providers database still should be fixed to make other aliases like @ya.ru work

@link2xt Actually the other domains are included in https://github.com/deltachat/provider-db/blob/master/_providers/yandex.ru.md .

**Comment by r10s** [comment]
hm, autoconfig directly from yandex reports that yandex._com_ should be used: https://autoconfig.yandex.com/mail/config-v1.1.xml

however, the autoconfig also reports that oauth2 should be used for authorisation - may it be that normal-login is not working at all? was there some change? unfortunately, my yandex test accounts are blocked or disabled, they want a mobile number now ... so i cannot really have a "quick look" ...

**Comment by westsibe** [comment]
![imapserver](https://user-images.githubusercontent.com/29564089/89729552-080d6580-da61-11ea-854b-e7a5a504c018.png)
![smtpserver](https://user-images.githubusercontent.com/29564089/89729560-22dfda00-da61-11ea-850e-8e9c75be7140.png)
^ My mail settings for @yandex.com in Aquamail for Android. All work.

Telephone number is optional (is not required.)

**Comment by r10s** [comment]
thanks for the information. as @link2xt pointed out, we should add server-settings to the provider-db of yandex - if possible, can you check if imap.yandex.com - works or not? this is the server DC is using as that is announced from yandex itself (as we do not have an entry in provider-db)

**Comment by link2xt** [comment]
I just checked, my Oauth2 setup for `@yandex.ru` is using smtp.yandex.com and imap.yandex.com in `configured_` settings. So for me oauth2 works with yandex as expected, their autoconfig is correct.

**Comment by westsibe** [comment]
> Delta Chat для Android.
> 
> imap.yandex.ru (входящий сервер) и smtp.yandex.ru (исходящий сервер) всегда для [login@ya.ru](mailto:login@ya.ru) , [login@yandex.ru](mailto:login@yandex.ru) , [login@yandex.com](mailto:login@yandex.com) и [login@yandex.ua](mailto:login@yandex.ua) !
> Никаких других smtp и imap серверов для Яндекс Почты!


> Delta Chat for Android.
> 
> imap.yandex.ru (incoming server) and smtp.yandex.ru (outgoing server) is always for [login@ya.ru](mailto:login@ya.ru), [login@yandex.ru](mailto:login@yandex.ru), [login@yandex.com](mailto:login@yandex.com) and [login@yandex.ua](mailto:login@yandex.ua)!
> No others smtp and imap servers for Yandex Mail!

I was wrong. Now works imap.ya.ru, imap.narod.ru, imap.yandex.com, smtp.ya.ru, smtp.yandex.com etc.

It's IMAP protocol was blocked by security reasons.in my other Yandex account. Sorry. I seen it just now. (Recommended close this ticket if no another bugs with Yandex Mail.)

**Comment by link2xt** [comment]
> It's IMAP protocol was blocked by security reasons.in my other Yandex account.

Real problem is that this error is not reported to the user: https://github.com/async-email/async-imap/issues/35

It still makes sense to add `smtp.yandex.com` and `imap.yandex.com` to provider database to speed up configuration, Yandex is very unlikely to change this as it will break all existing setups.

**Comment by r10s** [comment]
did a pr that ads yandex servers: https://github.com/deltachat/provider-db/pull/131

the real problem, https://github.com/async-email/async-imap/issues/35, however, stays open.

but there is nothing, we can do on that android-wise, however, we can keep this issue open to be reminded to the bug :)

**Comment by link2xt** [comment]
Let's close this bug then.

**Comment by avitex** [comment]
I've pushed a fix for async-email/async-imap#35 and drom my testing, [the PR](/async-email/async-imap/pull/36) fixes the issue.
If you'd like to give a spin:
```
async-imap = { git = "https://github.com/avitex/async-imap", branch = "fix-auth" }
```

**Comment by dumblob** [comment]
Merged 7 days ago into upstream async-imap. Yay! Thanks!
